main_two.py

```diff
@@ -0,0 +1,233 @@
+###################################
+############# Notice ##############
+###################################
+
+# Some parts need to be adapted later
+
+###################################
+########## import library #########
+###################################
+
+import numpy as np
+import pandas as pd
+
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+import torchvision.models as models
+from torchvision import transforms
+from torch.utils.data import Dataset, DataLoader
+import torch.optim as optim
+
+from sklearn import metrics
+
+#from skimage import io, color
+
+import time
+import os
+import pickle
+
+import matplotlib.pyplot as plt
+import scikitplot as skplt
+
+from train_valid import train, validation
+from Basic_CNN_Architecture import BasicCNN
+
+
+###################################
+########### load device ###########
+###################################
+
+# If there's a GPU available...
+if torch.cuda.is_available():
+
+    # Tell PyTorch to use the GPU.
+    device = torch.device("cuda")
+
+    print('There are %d GPU(s) available.' % torch.cuda.device_count())
+
+    print('We will use the GPU:', torch.cuda.get_device_name(0))
+
+# If not...
+else:
+    print('No GPU available, using the CPU instead.')
+    device = torch.device("cpu")
+
+
+###################################
+############ def path #############
+###################################
+
+# hpc
+# please change file path here
+train_df_path =
+val_df_path =
+test_df_path =
+
+root_dir =
+
+#batch size and epochs
+bs = 10
+epochs = 20
+
+
+###################################
+############ load data ############
+###################################
+
+# after feature engineering(task 3)
+
+train_loader =
+valid_loader =
+test_loader =
+
+###################################
+######### optim & criter ##########
+###################################
+
+optimizer = optim.Adam(model.parameters())
```

Comment:
I don't think we need to set this here, same with epoch count, etc. You wrote a function to take all those parameters so we can decide those later when we call the function.(Suppose we want SGD instead of Adam for instance) Same with the Criterions(97-104) below

Reply:
Sure!